src/components/QuranReader/TranslationView/TranslationText/index.tsx

```diff
@@ -155,11 +162,20 @@ const TranslationText: React.FC<Props> = ({
   const shouldShowFootnote = showFootnote && (footnote || isLoading);
   return (
     <div className={styles[`translation-font-size-${translationFontScale}`]} translate="no">
-      <div
-        onClick={(event) => onTextClicked(event)}
-        className={classNames(styles.text, styles[langData.direction], styles[langData.font])}
-        dangerouslySetInnerHTML={{ __html: text }}
-      />
+      {chapterName && reference ? (
+        <div className={classNames(styles.text, styles[langData.font])}>
+          &quot;{text}&quot;{' '}
+          <Link href={getVerseUrl(reference)} className={styles.referenceLink}>
```

Comment:
2 things:

1. We should use `getChapterWithStartingVerseUrl` instead of `getVerseUrl` because user can continue reading after this Ayah.
2. We should log clicking here.